### DISCUSSION TITLE
middleware call multiple times

### DISCUSSION THREAD

**Discussion by juyunsang** [fastapi/fastapi#8914] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [ ] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
logger = MyLogger()

def before():
    logger.update()
    logger.info('before')


@app.middleware("http")
async def test(request: Request, call_next):
    before()
    response = await call_next(request)
    return after(request, response)

async def after(request, response):
    logger.info('after')

```

### Description

Hello
Middleware is called multiple times when performance testing 
Intermittently, the following problems occur.
 - Before is called only once, but call_next and after are called multiple times
 - Call_next is not called, only after is called
 - 
I think it's right to be called before, call_next, after once.
Is there anything that I've implemented wrong?
### Environment

* OS: [e.g. Linux / Windows / macOS]: Linux
* FastAPI Version [e.g. 0.3.0]: 
* fastapi==0.65.2
uvicorn==0.14.0
fastapi-utils==0.2.1
python-multipart==0.0.5

To know the FastAPI version use:

```bash
python -c "import fastapi; print(fastapi.__version__)"
```

* Python version:

To know the Python version use:

```bash
python --version
```

**Comment by David-Lor** [comment, marked as answer]
Your middleware should return the response returned from "call_next". Your example is missing parts to make it work, so I run this, which is similar to what I understood you are trying to achieve:

```python
import uvicorn
from fastapi import FastAPI, Request

app = FastAPI()

def before():
    print("Before")

@app.middleware("http")
async def test(request: Request, call_next):
    before()
    response = await call_next(request)
    await after(request, response)
    return response

async def after(request, response):
    print("After")

@app.get("/status")
async def get_status():
    print("Request handler (a.k.a. call_next)")
    return "OK"

if __name__ == "__main__":
    uvicorn.run(app, port=5000)
```

It seems to work as expected:

```
$ python fastapi_basic.py 
INFO:     Started server process [29474]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:5000 (Press CTRL+C to quit)
Before
Request handler (a.k.a. call_next)
After
INFO:     127.0.0.1:51156 - "GET /status HTTP/1.1" 200 OK
```

**Comment by detronetdip** [comment]
Similar thing happened to me when i was starting the application with `uvicorn index:startServer --reload` and trying to hit an endpoint it triggers any middleware twice, but then i modified my code and make it like
```
from dotenv import load_dotenv
load_dotenv()

from app.app import startServer

 if __name__ == "__main__":
     import uvicorn
     uvicorn.run(startServer,port=3000)
``` 

after doing it I run my app with `python app.py` and then when i hit an endpoint it triggered the middleware only once
